core/src/main/java/org/geysermc/geyser/extension/GeyserExtensionLoader.java

```diff
@@ -157,16 +166,13 @@ protected void loadAllExtensions(@NonNull ExtensionManager extensionManager) {
                             return;
                         }
 
-                        // Completely different API version
-                        if (description.majorApiVersion() != Geyser.api().majorApiVersion()) {
-                            GeyserImpl.getInstance().getLogger().error(GeyserLocale.getLocaleStringLog("geyser.extensions.load.failed_api_version", name, description.apiVersion()));
-                            return;
-                        }
-
-                        // If the extension requires new API features, being backwards compatible
-                        if (description.minorApiVersion() > Geyser.api().minorApiVersion()) {
-                            GeyserImpl.getInstance().getLogger().error(GeyserLocale.getLocaleStringLog("geyser.extensions.load.failed_api_version", name, description.apiVersion()));
-                            return;
+                        // Completely different API version - or if the extension requires new API features, being backwards compatible
+                        if (GeyserApi.api().geyserApiVersion().isCompatible(new ApiVersion(description.majorApiVersion(), description.minorApiVersion(), description.patchApiVersion()))) {
+                            // workaround for the switch to Geyser API version
+                            if (description.majorApiVersion() != 1) {
+                                GeyserImpl.getInstance().getLogger().error(GeyserLocale.getLocaleStringLog("geyser.extensions.load.failed_api_version", name, description.apiVersion()));
+                                return;
+                            }
```

Comment:
I think you're missing a `!` for the first if-statement?

Also maybe add a comment about removing the workaround once the major version of the base-api is bumped